examples/diff.c

```diff
@@ -47,6 +47,15 @@ enum {
 	CACHE_NONE = 2
 };
 
+#define DEFAULT_THRESHOLD 50
+#define DEFAULT_BREAK_REWRITE_THRESHOLD 60
+#define USE_DEFAULT(X) ((X) == 0 || (X) > 100)
+
+struct pair {
+	int first;
+	int second;
+};
```

Comment:
The way we do it in the library is to return 0 for success and a negative number for error, which is common is unix systems. The purpose of a `die()` function is to show the user an error and then exiting the program with an appropriate error code. You should show the error to the user, which AFAICT is more than this does right now.